extensions/sql-database-projects/src/models/project.ts

```diff
@@ -307,22 +307,31 @@ export class Project implements ISqlProject {
 	/**
 	 * Adds a folder to the project, and saves the project file
 	 * @param relativeFolderPath Relative path of the folder
+	 * @param ignoreDuplicate
+	 *	Flag that indicates whether duplicate entries should be ignored. If flag is set to `true` and item already
+	 *	exists in the project file, then existing entry will be returned.
 	 */
-	public async addFolderItem(relativeFolderPath: string): Promise<FileProjectEntry> {
+	public async addFolderItem(relativeFolderPath: string, ignoreDuplicate?: boolean): Promise<FileProjectEntry> {
```

Comment:
I was debating whether to use default value or `boolean | undefined` type. It appears that `.. | undefined` is the preferred way in this codebase, so I followed that practice.